pipeline/ValidateTrees.C

```diff
@@ -1,52 +1,60 @@
 #include <TFile.h>
 #include <TTree.h>
+
 #include <iostream>
 #include <vector>
-Int_t ValidateTrees(TString validationFile) {
-    TFile* f = new TFile("results.root");
-    TTree* tr = (TTree*)f->Get("bTree");
-    std::vector<double>* vr = 0;
-    tr->SetBranchAddress("doubleObservables", &vr);
-
-    TFile* fV = new TFile(validationFile);
-    TTree* tV = (TTree*)fV->Get("bTree");
-    std::vector<double>* vV = 0;
-    tV->SetBranchAddress("doubleObservables", &vV);
-
-    if (tr->GetEntries() != tV->GetEntries()) {
+
+Int_t ValidateTrees(const char* validationFilename, const char* inputFilename = "results.root") {
+    TFile inputFile(inputFilename);
+    TTree* inputTree = inputFile.Get<TTree>("bTree");
+    std::vector<double>* inputObservableVector = 0;
+    inputTree->SetBranchAddress("doubleObservables", &inputObservableVector);
+
+    TFile validationFile(inputFilename);
```

Comment:
I believe this should be `TFile validationFile(validationFilename);`

Reply:
> I believe this should be `TFile validationFile(validationFilename);`

good catch thanks, its now fixed.